src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs

```diff
@@ -528,13 +524,42 @@ public override AssemblyName GetName(bool copiedName)
             return AssemblyName.Create(_mono_assembly, null);
         }
 
-        // FIXME: "This always returns an empty array"
         public override AssemblyName[] GetReferencedAssemblies()
         {
-            throw new NotImplementedException();
-#if FALSE
-			return GetReferencedAssemblies (this);
-#endif
+            //copied from RuntimeAssembly.cs
+             using (var nativeNames = new Mono.SafeGPtrArrayHandle(RuntimeAssembly.InternalGetReferencedAssemblies(this)))
```

Comment:
What do you think about keeping `InternalGetReferencedAssemblies` `private` to `RuntimeAssembly` and instead adding this safe wrapper as an `internal` method to that class?

ie: `AssemblyBuilder.Mono.cs` would just have `GetReferencedAssemblies () => RuntimeAssembly.GetReferencedAssemblies (this)`

and `RuntimeAssembly.cs` would have `internal static Assembly[] GetReferencedAssemblies (Assembly assm) { ... }` which is a wrapper around `InternalGetReferencedAsseemblies` ?

Reply:
Yeah sounds good, it wasn't clear from the beginning that the copied code would be unmodified but I think it is.